test/jdk/com/sun/jdi/JdwpNetProps.java

```diff
@@ -67,19 +67,26 @@ public static void main(String[] args) throws Exception {
             new ListenTest("localhost", ipv4Address)
                     .preferIPv4Stack(false)
                     .run(TestResult.Success);
+            new ListenTest("localhost", ipv4Address)
+                    .preferIPv4Stack(true)
+                    .preferIPv6Addresses("system")
+                    .run(TestResult.Success);
```

Comment:
Does this test guarantee that "system" was honored, or only verify that there was no failure to attach? In other words, wouldn't this pass even if the debug agent ignored the use of  "system"?

What happens if using "system" with `preferIPv4Stack(false)`? I'm not sure why you don't have a test case for this. I guess maybe the reason is we don't know if the test case would pass or not because we don't know if "system" will result in using v4 (pass) or v6 (fail). Is there a way to determine that in advanced?

Reply:
> Does this test guarantee that "system" was honored, or only verify that there was no failure to attach? In other words, wouldn't this pass even if the debug agent ignored the use of "system"?

That is true, with `preferIPv4Stack(true)` I don't think there's a decision to make about ordering. Would you like this test case to be removed?

> What happens if using "system" with `preferIPv4Stack(false)`? I'm not sure why you don't have a test case for this.

I have added additional test cases.

> I guess maybe the reason is we don't know if the test case would pass or not because we don't know if "system" will result in using v4 (pass) or v6 (fail). Is there a way to determine that in advanced?

Right, it depends on what `getaddrinfo` is doing on the current system. One way to determine that in advance is to run the test with `-Djava.net.preferIPv6Address=system` and check which address `InetAddress.getAllByName("localhost")` returns first, and then assert that is consistent with what JDWP is doing. I'm didn't see a more direct way to check what `getaddrinfo` is going to do, or avoid relying on its behavior on the current system.